### ISSUE TITLE
[Feature] Add in recall metric for Yolov4

### ISSUE THREAD

**Issue by anthonyfong100** [whizzmobility/models#87]
# Prerequisites

Please answer the following question for yourself before submitting an issue.

- [X] I checked to make sure that this feature has not been requested already.

## 1. The entire URL of the file you are using

https://github.com/tensorflow/models/tree/master/official/...

## 2. Describe the feature you request
Add in recall metrics for Yolov4 --> this lets us know how much % of time we are misssing the junctions / road breaks / humans etc

## 3. Additional context
Ideally when building the system at the start --> we can make it safer by optimizing for recall --> can tune the prediction confidence threshold

**Comment by rehohoho** [comment]
Oops, the implementation is wrong, it does
1) Filter by iou
2) Accumulate confusion matrix

It should be 
```
for each detection that has a confidence score > threshold:

  among the ground-truths, choose one that belongs to the same class and has the highest IoU with the detection
  
  if no ground-truth can be chosen or IoU < threshold (e.g., 0.5):
    the detection is a false positive
  else:
    the detection is a true positive
```
taken from https://blog.zenggyu.com/en/post/2018-12-16/an-introduction-to-evaluation-metrics-for-object-detection/

**Comment by rehohoho** [comment]
Closed by #89.
I do observe that the recall metrics are very low, but this is because the confidence scores are very low.
See #92 if we want to ensure we use the exact same methods as COCO detection set.
